lua/starfall/libs_cl/bass.lua

```diff
@@ -86,17 +163,49 @@ function bass_library.loadFile(path, flags, callback)
 				snd:Stop()
 				plyCount:free(instance.player, 1)
 			else
-				sounds[snd] = true
+				table.insert(instanceSounds, snd)
+				soundDatas[snd] = { -- IGModAudioChannel is userdata, so we can't attach extra key-value pairs or functions to it directly.
```

Comment:
Would be cool if this was a class instead so all the local functions could be class methods.

Reply:
Maybe, but I feel like that would only add unnecessary layers of wrapping and indexing.